tests/serverpod_test_server/test_e2e/basic_types_test.dart

```diff
@@ -148,4 +148,20 @@ void main() {
     var result = await client.basicTypes.testUri(null);
     expect(result, isNull);
   });
+
+  test(
+      'Given the test server, when a `BigInt`` is sent to the server, then it is returned verbatim',
+      () async {
+    var bigInt = BigInt.parse('-12345678901234567890');
+
+    var result = await client.basicTypes.testBigInt(bigInt);
+    expect(result, equals(bigInt));
+  });
```

Comment:
Suggestion: I just realized that we have nothing like this for streams! The streaming serialization is also using a different code path for serialization under the hood so it is relevant to replicate these tests for them!

This means duplicating this file and test cases but using streams instead, there are not streaming endpoints like this today in the project!

An echo stream endpoint could be implemented like this:
```dart
Stream<int?> testInt(Session session, Stream<int?> value) async* {
    await for (var v in value) {
      yield v;
    }
  }
```

Reply:
Got it, will add that :)

Reply:
Added test for the basic types streaming, so we have them all covered.

There were a few cases with a handful of types in https://github.com/tp/serverpod/blob/6d209b322398548cd4abd70f3d8f407018073253/tests/serverpod_test_server/test_e2e/method_streaming, but not all of them covered.

So good that you thought of that, so that we also have new types like `Duration`, `BigInt`, and `Uri` covered :)